Sources/LanguageServerProtocol/SupportTypes/WorkspaceSettings.swift

```diff
@@ -16,13 +16,16 @@
 public enum WorkspaceSettingsChange: Codable, Hashable {
 
   case clangd(ClangWorkspaceSettings)
+  case scheme(BuildScheme)
```

Comment:
Could we simplify this and make it a bit clearer by doing

```swift
case sourcekitlsp(SourceKitLSPWorkspaceSettings)
...
struct SourceKitLSPWorkspaceSettings: Codable {
  var scheme: BuildScheme? = nil
}
```

This would give us room to add other kinds of settings.  It also lets us put the "sourcekit-lsp" coding key namespace in one place instead of duplicating it if we ever add more settings.

Reply:
Good point, added setting wrapper model